### PR TITLE
10.0-Multiple website_sale* fixes

### PR DESCRIPTION
**Pull request by em230418**

### CODE REVIEW

**Comment by yelizariev** [comment]
For test fixing use :shield:

**Comment by em230418** [comment]
Should I rename this pr? If yes, any suggestions?

File: website_sale_search_tags/models.py [hunk 1]
```diff
@@ -8,8 +8,13 @@ class Product(models.Model):
...
                 ('tag_ids', 'ilike', self.env.context.get('search_tags')) not in domain):
-            domain.insert(domain.index(('sale_ok', '=', True)) + 1, '|')
-            domain += [('tag_ids', 'ilike', self.env.context.get('search_tags'))]
+
+            i = domain.index(('sale_ok', '=', True)) + 1
+            while type(domain[i]) in [tuple, list] and i < len(domain):
+                i = i + 1
+
+            domain.insert(i, '|')
+            domain.insert(i + 1, ('tag_ids', 'ilike', self.env.context.get('search_tags')))
```
Comment (yelizariev) [inline website_sale_search_tags/models.py]: Update version and changelog file

File: website_sale_search_tags/models.py [hunk 2]
```diff
@@ -8,8 +8,13 @@ class Product(models.Model):
     def _extend_domain(self, domain):
         if (self.env.context.get('search_tags') and
                 ('tag_ids', 'ilike', self.env.context.get('search_tags')) not in domain):
-            domain.insert(domain.index(('sale_ok', '=', True)) + 1, '|')
-            domain += [('tag_ids', 'ilike', self.env.context.get('search_tags'))]
+
+            i = domain.index(('sale_ok', '=', True)) + 1
+            while type(domain[i]) in [tuple, list] and i < len(domain):
+                i = i + 1
```
Comment (yelizariev) [inline website_sale_search_tags/models.py]: Consider to add a comment what is going on here

**Comment by yelizariev** [comment]
Next time use following format: 10.0-FEATURENAME


> Should I rename this pr? If yes, any suggestions?

**Comment by yelizariev** [comment]
The PR name is up to you

File: website_sale_search_tags/models.py (hunk 1 above)
Comment (em230418) [reply website_sale_search_tags/models.py]: Should I write
`10.0.1.0.3` or `1.0.3` in changelog.rst?

File: website_sale_search_tags/models.py (hunk 2 above)
Comment (em230418) [reply website_sale_search_tags/models.py]: > Consider to add a comment what is going on here

Done

File: website_sale_search_tags/models.py (hunk 1 above)
Comment (yelizariev) [reply website_sale_search_tags/models.py]: 1.0.3

Comment (em230418) [reply website_sale_search_tags/models.py]: Done https://github.com/it-projects-llc/website-addons/pull/429/files#diff-21f07e4f2b90527d2a47926e05ce6483

File: website_sale_search_tags/models.py [hunk 3]
```diff
@@ -8,8 +8,14 @@ class Product(models.Model):
     def _extend_domain(self, domain):
         if (self.env.context.get('search_tags') and
                 ('tag_ids', 'ilike', self.env.context.get('search_tags')) not in domain):
-            domain.insert(domain.index(('sale_ok', '=', True)) + 1, '|')
-            domain += [('tag_ids', 'ilike', self.env.context.get('search_tags'))]
+
+            # calculating position, after which domain new domains must be placed
```
Comment (yelizariev) [inline website_sale_search_tags/models.py]: Could you explain in comment about how the code works, some examples maybe (original, after the modification). The code a bit complicated and need some explanation

Comment (em230418) [reply website_sale_search_tags/models.py]: I will add test cases for `_extend_domain`

Comment (em230418) [reply website_sale_search_tags/models.py]: Done: https://github.com/it-projects-llc/website-addons/pull/429/commits/3f8a0129862ebb8b52184daf396e068e223d307a

**Pull request merged**